river/Config.zig

```diff
@@ -85,10 +86,27 @@ repeat_rate: u31 = 25,
 /// Keyboard repeat delay in milliseconds
 repeat_delay: u31 = 600,
 
+/// Xkb state for keysym-to-keycode translation of `to-code` mappings
+translation_state: *xkb.State,
+
 pub fn init() !Self {
+    const rules = xkb.RuleNames{
+        .rules = null,
+        .model = null,
+        .layout = null,
+        .variant = null,
+        .options = null,
+    };
+    const context = xkb.Context.new(.no_flags) orelse return error.XkbContextFailed;
+    defer context.unref();
+    const keymap = xkb.Keymap.newFromNames(context, &rules, .no_flags) orelse return error.XkbKeymapFailed;
```

Comment:
Passing null here is treated the same as passing a all-null `xkb.RuleNames` struct.
```suggestion
    const keymap = xkb.Keymap.newFromNames(context, null, .no_flags) orelse return error.XkbKeymapFailed;
```